packages/cf-deploy-config-inquirer/src/prompts/prompts.ts

```diff
@@ -85,21 +79,39 @@ async function getApplicationRouterPrompt(
     } as ConfirmQuestion<CfDeployConfigAnswers>;
 }
 
+/**
+ *
+ * @param addOverwriteQuestion Indicates whether the overwrite question should be shown.
+ * @returns A confirmation question object which overwrites destination.
+ */
+function getOverwritePrompt(addOverwriteQuestion: boolean): CfDeployConfigQuestions {
+    return {
+        type: 'confirm',
+        name: 'cfOverwrite',
+        guiOptions: {
+            hint: t('prompts.overwriteHintMessage')
+        },
+        default: () => {
+            return true;
+        },
+        message: (): string => t('prompts.overwriteMessage'),
+        when: (): boolean => showOverwriteQuestion(addOverwriteQuestion)
```

Comment:
Can you please put `when` conditions as the first property and inline if its a basic conditional check.